### PR TITLE
add test config

### PR DESCRIPTION
**Pull request by liuyejia**
tests

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+    - name: Set up Python
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    # - name: Install dependencies
```
Comment (dovahcrow) [inline .github/workflows/test.yml]: I think you can just install the `dataprep` from PyPI

File: .github/workflows/runTests.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
+echo $PWD
```
Comment (dovahcrow) [inline .github/workflows/runTests.sh]: This file is duplicated

File: .github/runTests.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
+echo $PWD
+for file in $1
+do
+        echo $file
+        filename="$(basename -- $file)"
+        extension="${filename##*.}"
+        parentdir="$(basename "$(dirname "$file")")"
+        echo parent_dir:${parentdir}
+        if [[ "$parentdir" =~ ^(dblp|spotify|twitter|yelp)$ ]]; then
```
Comment (dovahcrow) [inline .github/runTests.sh]: Could it be more general? Currently, this requires us to change the script every time when there's a new config coming in.

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,59 @@
+[tool.poetry]
```
Comment (dovahcrow) [inline pyproject.toml]: This file is not required anymore if we use PyPI.

**Review by dovahcrow — CHANGES_REQUESTED** [review]

File: .github/runTests.sh [hunk 5]
```diff
@@ -0,0 +1,17 @@
+echo $PWD
+for file in $1
```
Comment (dovahcrow) [inline .github/runTests.sh:3]: What if there're multiple config files in one `parentdir`?

**Review by dovahcrow — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (liuyejia) [reply .github/workflows/test.yml]: updated and changed to "pip install dataprep"

File: .github/workflows/runTests.sh (hunk 2 above)
Comment (liuyejia) [reply .github/workflows/runTests.sh]: Good catch! Thanks! Removed

File: .github/runTests.sh (hunk 3 above)
Comment (liuyejia) [reply .github/runTests.sh]: changed to search subdirectories under root of repo

File: pyproject.toml (hunk 4 above)
Comment (liuyejia) [reply pyproject.toml]: removed

File: .github/runTests.sh (hunk 5 above)
Comment (liuyejia) [reply .github/runTests.sh:3]: add code to filter out unique changed directories

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+        pip install pytest
+    
+    - name: Run tests
+      run: |
+        URL="https://api.github.com/repos/${GITHUB_REPOSITORY}/pulls/${{ github.event.pull_request.number }}/files"
+        FILES=$(curl -s -X GET -G $URL | jq -r '.[] | .filename')
+        chmod +x .github/runTests.sh
+        .github/runTests.sh "$FILES"
+    
+
```
Comment (dovahcrow) [inline .github/workflows/test.yml]: Remove these newlines

File: .github/workflows/test.yml [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    - name: Install dependencies
+      run: |
+        pip install dataprep
+        pip install pytest
+    
+    - name: Run tests
+      run: |
+        URL="https://api.github.com/repos/${GITHUB_REPOSITORY}/pulls/${{ github.event.pull_request.number }}/files"
+        FILES=$(curl -s -X GET -G $URL | jq -r '.[] | .filename')
+        chmod +x .github/runTests.sh
```
Comment (dovahcrow) [inline .github/workflows/test.yml]: Remove this line, change the permission in the commit.

**Review by dovahcrow — COMMENTED** [review]
Other things pending for merge:

* Rename: test.yml -> config_test.yml
* Remove `dblp/exp.py` and `dblp/tests/*`
* Squash the commits to a single one

File: .github/workflows/test.yml (hunk 7 above)
Comment (liuyejia) [reply .github/workflows/test.yml]: removed and changed permission in commits

File: .github/workflows/test.yml (hunk 6 above)
Comment (liuyejia) [reply .github/workflows/test.yml]: removed

**Pull request merged**